src/mongocxx/lib/mongocxx/v_noabi/mongocxx/write_concern.cpp

```diff
@@ -175,6 +175,9 @@ bsoncxx::v_noabi::document::value write_concern::to_document() const {
                 if (auto t = tag()) {
                     doc.append(kvp("w", *t));
                 }
+                break;
+
+            case write_concern::level::k_acknowledged:
```

Comment:
> is the write_concern::level::k_acknowledged do-nothing fallthrough in write_concern.cpp correct?

I checked out the [API reference ](https://mongocxx.org/api/current/classmongocxx_1_1v__noabi_1_1write__concern.html#a756cc9e4f51467924887b2ceda9c8856)and the MongoDB docs, not sure myself.

Reply:
I expect this case is unreachable. If this `else` block is entered, then `nodes()` did not return a value. That implies `acknowledge_level()` cannot be `k_acknowledged`.

I expect this could assert:

```c++
            case write_concern::level::k_acknowledged:
                // Acknowledged write concern is expected to return `nodes()` above.
                MONGOCXX_UNREACHABLE;
```